### DISCUSSION TITLE
Open-loop test.

### DISCUSSION THREAD

**Discussion by QusayHawari** [OpenFAST/openfast#1012] [Q&A]
Hi, 

I am trying to perform an open-loop analysis on the UMaine IEA15MW turbine model using OpenFAST. 

Do you think the right way to do it is to deactivate the controller in ServoDyn by setting PCMode and VSContrl to zero, and assigning a certain blade pitch as an initial value in ElastoDyn?

Do you think it's also possible to enter an array of blade pitch commands as an initial condition instead of one value?

Thanks,
Qusay.

**Comment by jjonkman** [comment]
Dear @QusayHawari,

You can disable active pitch control by setting `PCMode` = 0 in ServoDyn (and by ensuring that the override pitch maneuvers are not engaged during the simulation).  In this case, the initial pitch set in ElastoDyn will remain constant throughout the simulation.

Setting `VSContrl` = 0 in ServoDyn implies that a generator model is specified instead.  Presumably by "open-loop analysis" you want to specify constant torque; is that correct?   To set constant torque in ServoDyn, you can set `VSContrl` = 1, `VS_RtTq` = to the constant torque you want, and `VS_RtGnSp` = `VS_Rgn2K` = `VS_SlPc` = some small number greater than zero, e.g., 0.001.

Best regards,

**Reply by QusayHawari** [reply]
Hi, 

Thanks for your reply. 

I am trying to test the open loop response under turbulent speed (with a mean of 20m/s) by applying a fixed blade pitch (from ElastoDyn) equal to the operating blade pitch point at 20m/s. 

Perhaps another way to do it is to keep PCMode set at 5, while changing Ki=0 and Kp =1 (all 28 of them) in the DISCON file. This will basically replace the PI pitch controller with a gain of 1 at all above rated wind speeds. 

Qusay.

**Reply by jjonkman** [reply]
Hi Qusay,

The DISCON controller for the IEA 15-MW reference wind turbine atop the UMaine semi controls both torque and pitch.  If you desire constant pitch and constant torque, it is easiest to do as I describe in my prior post.

Best regards,

**Reply by QusayHawari** [reply]
Hello Jason,


I performed the open-loop test as you suggested by setting PCmode =0 and VSContrl = 1. I initialized the blade pitch to be at the operating point of 20 m/s (I am applying turbulent wind with the same mean). 

I then tested the closed-loop response with the same inputs and initial condition of the blade pitch except that I set PCMode = 5 to use the DISCON file for gain scheduling and other control commands. 

The response of both open and closed loops seem to be similar. Does this mean that even if PCMode = 0, there is an internal controller switched on?

![OL_CL_Turb_15MW](https://user-images.githubusercontent.com/91909348/155592532-97ef84a5-67ed-4beb-961a-069fb23b54f7.jpg)

QUSAY.

**Reply by jjonkman** [reply]
Hi @QusayHawari,

There is no internal pitch controller active when `PCMode` = 0.  I can't really comment on your results without knowing more about your simulation set up.

Best regards,

**Reply by QusayHawari** [reply]
Hi,, 

Yes, sorry about that. I have attached the .fst, ServoDyn, ElastoDyn, and HydroDyn files. Please have a look whenever you have time.

[Simulation_Setup.zip](https://github.com/OpenFAST/openfast/files/8135506/Simulation_Setup.zip)

Qusay.

**Reply by jjonkman** [reply]
What type of wind are you using?  Is it turbulent?  What do the blade-pitch and generator torque outputs look like?

Best regards,

**Reply by QusayHawari** [reply]
Hello Jason,

I' am using turbulent wind speed with a mean value of 20m/s. For the blade pitch and generator torque they should be constants since VSContrl is set at 1, and PCMode is set at 0 with an initial value of 17.9 degs for the blade pitch. Indeed they are constant looking at the results.
![Wind_Speed](https://user-images.githubusercontent.com/91909348/155724606-b757d8eb-7339-491f-96e6-1aa8a7fcc98d.jpg)
![Gen_Tq](https://user-images.githubusercontent.com/91909348/155724621-91ad1635-8563-4c5a-b140-5cc3b4aed9e7.jpg)
![Bld1](https://user-images.githubusercontent.com/91909348/155724745-67fe74bb-42fa-45ae-8a01-8477860bd126.jpg)

Qusay

**Reply by jjonkman** [reply]
And how do these plots compare between the open- and closed-loop response?

**Reply by QusayHawari** [reply]
Hi,


Here they are:

![Bld1_CL](https://user-images.githubusercontent.com/91909348/155725546-994aaecc-2a14-45b9-ab21-bf321561ebd3.jpg)
![Gen_Tq_CL](https://user-images.githubusercontent.com/91909348/155725562-56545991-37f3-4f8a-b5f2-c6ec86d3fbf8.jpg)


Qusay.

**Reply by jjonkman** [reply]
It looks like the response is similar between the two because the control actions are not too different between the two after the start-up transient.  It appears that the closed-loop controller you are using uses constant torque in Region 3, just as in open loop.

**Reply by QusayHawari** [reply]
Yes, I agree, Since the objective of the pitch controller is to stabilize the blade pitch at the operating point of 20m/s, I am essentially doing the same thing in open-loop when I apply that objective as a form of initial value for the pitch. In both cases the torque is constant. 

Thank you so much for your FAST and helpful replies :)

Qusay.
